### ISSUE TITLE
[NEWS] Wysyłam z Allegro: od 22 sierpnia 2024 dla usług dostawy Allegro Standard nie skorzystasz już z umowy własnej / Ship with Allegro: as of August 22, 2024, you will no longer use the custom agreement for Allegro Standard delivery services

### ISSUE THREAD

**Issue by AureliuszBrussy** [allegro/allegro-api#9625]
Od **22 sierpania 2024 roku** podczas nadania przesyłki z dostawą [w ramach Allegro Standard](https://help.allegro.com/sell/pl/a/jak-rozliczac-koszty-dostaw-na-allegro-Agbjkv5B7U7#jakie-metody-dostawy-zaliczamy-do-grupy-metod-rozliczanych-sposobem-allegro-standard), sprzedający nie będzie mógł już skorzystać z umowy własnej. Przesyłki takie nada wyłącznie w ramach umowy Allegro.

Ponadto, dla usług dostawy **spoza Allegro Standard**, dla których sprzedający posiada umowy własne, nie będzie mógł już zmienić przewoźnika realizującego dostawę. Oznacza to, że powinien utworzyć przesyłkę zgodnie z wybranym przez kupującego w zamówieniu przewoźnikiem. 

**Jak rozpoznać usługę Allegro Standard?**

Gdy odpytasz o dostępne usługi dostawy za pomocą [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation#operation/getDeliveryServices), dla usługi Allegro Standard, w polu "**owner**" zwrócimy wartość "**ALLEGRO**". 

Tworząc przesyłkę ([POST /shipment-management/shipments/create-commands](https://developer.allegro.pl/documentation#operation/createNewShipment)) z metodą dostawy Allegro Standard, w polu "**credentialsId**" nie przekazuj żadnej wartości lub przekaż **null**.

Od **22 sierpnia**, gdy spróbujesz utworzyć przesyłkę niezgodnie z nowymi wytycznymi - zwrócimy odpowiedni komunikat błędu:

```json5
{
  code: VALIDATION_ERROR,
  path: „deliveryMethodId”,
  message: „Cannot create a shipment using own contract. Method assigned to this order is the Allegro Standard one. Change method and try again.”
}
```

Więcej o **Wysyłam z Allegro** przeczytasz [w naszym poradniku](https://developer.allegro.pl/tutorials/jak-zarzadzac-przesylkami-przez-wysylam-z-allegro-LRVjK7K21sY).
___________

From **August 22, 2024**, when shipping with [Allegro Standard delivery](https://help.allegro.com/sell/en/a/how-to-settle-delivery-costs-on-allegro-K6V3vAGObFj#what-delivery-options-we-settle-within-allegro-standard), the seller will no longer be able to use a custom agreement. The seller will only send such shipments under the Allegro contract.

In addition, for delivery services **outside of Allegro Standard**, for which the seller has his own agreements, he will no longer be able to change the carrier handling the delivery. This means that the seller should create the shipment following the carrier selected by the buyer in the order. 

**How to recognize the Allegro Standard service?**

When you check out available delivery services using [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation#operation/getDeliveryServices), we will return the "**ALLEGRO**" value in the "**owner**" field for Allegro Standard services. 

When creating a shipment ([POST /shipment-management/shipments/create-commands](https://developer.allegro.pl/documentation#operation/createNewShipment)) with the Allegro Standard delivery method, in the "**credentialsId**" field, do not pass any value or pass **null**.

From **August 22**, when you try to create a shipment that does not comply with the new guidelines - we will return the appropriate error message:

```json5
{
  code: VALIDATION_ERROR,
  path: „deliveryMethodId”,
  message: „Cannot create a shipment using own contract. Method assigned to this order is the Allegro Standard one. Change method and try again.”
}
```

You can read more about **Ship from Allegro** [in our guide](https://developer.allegro.pl/tutorials/how-to-manage-parcels-via-ship-with-allegro-ZM9YAyGKWTV).

**Comment by jesiiu** [comment]
Czy dotyczy to również inpostu? 
Zasób /shipment-management/delivery-services zwraca taką odpowiedź (wzięte z dokumentacji allegro) 
"services": [
{
"id": {
"deliveryMethodId": "c3066682-97a3-42fe-9eb5-3beeccab840c",
"credentialsId": "c9e6f40a-3d25-48fc-838c-055ceb1c5bc0"
},
Teraz tworząc etykietę przekazuję deliveryMethodId oraz credentialsId, po tej akutalizacji nie będzie można utworzyć etykiety w momencie w którym będzie przekazane credentialsId?

**Comment by AureliuszBrussy** [comment]
Nie, InPost nie należy do usług z grupy Allegro Standard. Metody dostawy Allegro Standard są wymienione [w pomocy](https://help.allegro.com/sell/pl/a/jak-rozliczac-koszty-dostaw-na-allegro-Agbjkv5B7U7#jakie-metody-dostawy-zaliczamy-do-grupy-metod-rozliczanych-sposobem-allegro-standard), w komunikacie także podałem link.

**Comment by AureliuszBrussy** [comment]
Już teraz dla usług Orlen sprzedający ma dostęp do umowy, której właścicielem jest Allegro. Możecie w dowolnym momencie zacząć z niej korzystać. Od 22 sierpnia, jeżeli kupujący wybierze jedną z metod dostawy Allegro Standard, nadasz ją tylko na umowie Allegro. 

InPostu ta sytuacja faktycznie nie dotyczy, gdyż ta usługa dostawy nie jest częścią Allegro Standard.

**Comment by AureliuszBrussy** [comment]
Nie, jeżeli klient złoży zamówienie z dostawą Allegro Standard, np. **Allegro Odbiór w Punkcie ORLEN Paczka**, to będziesz mógł utworzyć paczkę tylko za pomocą umowy Allegro, nie będziesz mógł skorzystać z umowy klienta. Sprzedający wciąż może udostępniać w swoim cenniku dostawy metodę "ORLEN Paczka", czyli taką nie będącą częścią Allegro Standard. Wtedy, jeżeli kupujący wybierze tą metodę dostawy, wciąż nadasz jak na umowie sprzedającego.

**Comment by AureliuszBrussy** [comment]
Ustalam to. Wrócę z odpowiedzią.

**Comment by AureliuszBrussy** [comment]
@ZAK28ZAK wciąż będziesz mógł swobodnie nadawać przesyłki za pomocą API DPD. Nic się dla Ciebie nie zmieni.

**Comment by AureliuszBrussy** [comment]
@Ewkaaa92 sprawdź, czy w [cennikach dostawy](https://salescenter.allegro.com/delivery-settings) masz udostępnione odpowiednie metody dostawy, czyli te wliczające się do Allegro Standard (czyli te z dopiskiem Allegro ORLEN). Wygląda na to, że udostępniasz w cenniku dostawy kupującym metody dostawy spoza umowy Allegro i próbujesz dla tych zamówień (gdzie kupujący wybrał taką metodę dostawy) wygenerować przesyłkę w ramach umowy Allegro.

**Comment by Ewkaaa92** [comment]
W cennikach dostawy mam udostępnione Allegro Delivery:

![image](https://github.com/user-attachments/assets/5b5437f0-2166-46a6-b092-6c0a3e7b2f78)

Nie mam podpiętej żadnej umowy własnej.

**Comment by Ewkaaa92** [comment]
traceid: 35f98c01d2a59050

{"commandId":"6a0d910e-376a-4157-9adb-f904fa382b39","status":"ERROR","errors":[{"code":"DELIVERY_METHOD_NOT_AVAILABLE","message":"Cannot create a shipment using the Allegro Standard contract. Delivery method assigned to this order is the merchant controlled one. Change the service and try again.","details":"Delivery Method is not available for this order","path":"deliveryMethodId","userMessage":"Nie można utworzyć przesyłki przy użyciu umowy Allegro Standard. Metoda przypisana do tego zamówienia to umowa własna. Zmień umowę i spróbuj ponownie."}],"shipmentId":null}',
)

**Comment by AureliuszBrussy** [comment]
@Ewkaaa92 problem polega na niepoprawnym zmapowaniu metody dostawy z uslugą dostawy. Zwróć uwagę na to, jaki `id` metody dostawy zwracamy w zamówieniu (pole `delivery.method.id`). Takie samo `id` usługi dostawy musisz wybrać, generując przesyłkę. Próbujesz natomiast wykorzystać inną usługę.

**Comment by Ewkaaa92** [comment]
Wysłałam Pana odpowiedź do supportu aplikacji, poniżej odpowiedź:

Jak rozumiem chodzi o deliveryMethodId, wystawiamy 025cb739-ef8e-45c4-9e7b-ce4e7fece8be (metoda shipment-management/shipments/create-commands)

Otrzymujemy z Allegro

 array (
    'commandId' => '2d9b32bc-50f1-4c78-b2c8-e550d489fabd',
    'input' => 
    array (
      'deliveryMethodId' => '025cb739-ef8e-45c4-9e7b-ce4e7fece8be',

.....

Jak widać jest to ta sama metoda (id)

Następnie pytamy sie o 

shipment-management/shipments/create-commands/2d9b32bc-50f1-4c78-b2c8-e550d489fabd
 
i tu dostajemy ten błąd

**Comment by Ewkaaa92** [comment]
Nasz support sprawdza jeszcze jedną kwestię, także dam znać :)
